pandas_gbq/gbq.py

```diff
@@ -170,6 +170,9 @@ def __init__(self, project_id, reauth=False,
         from google.api_core.exceptions import GoogleAPIError
         from google.api_core.exceptions import ClientError
         self.http_error = (ClientError, GoogleAPIError)
+        if not project_id:
+            from google.auth import default
+            _, project_id = default()
```

Comment:
On second thought, this could cause us to go through the whole "application default credentials" flow twice, which could be quite slow. (It checks for env var, checks for GCE, checks for App Engine, ...)

Reply:
Could we change https://github.com/maxim-lian/pandas-gbq/blob/a7f6c43434bc21e1d2fc7633d3c5cb9c04060a4a/pandas_gbq/gbq.py#L223 to not discard the project instead?

Reply:
Yeah not great:
```

In [4]: %timeit google.auth.default()
573 ms ± 14.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
What do you think about changing those methods to return 2-tuples?

Reply:
I think tuples would be reasonable. The other (more disruptive, probably) option would be to pass in the project to the credentials methods and have them return a `google.cloud.bigquery.Client`.

Reply:
I updated this PR to tuples as we discussed.